To: vim_dev@googlegroups.com
Subject: Patch 7.3.232
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.232
Problem:    Python doesn't compile without +multi_byte
Solution:   Use "latin1" when MULTI_BYTE is not defined.
Files:      src/if_py_both.h


*** ../vim-7.3.231/src/if_py_both.h	2011-06-19 00:27:46.000000000 +0200
--- src/if_py_both.h	2011-06-26 03:58:07.000000000 +0200
***************
*** 12,17 ****
--- 12,23 ----
   * Common code for if_python.c and if_python3.c.
   */
  
+ #ifdef FEAT_MBYTE
+ # define ENC_OPT p_enc
+ #else
+ # define ENC_OPT "latin1"
+ #endif
+ 
  /*
   * obtain a lock on the Vim data structures
   */
***************
*** 68,74 ****
      char *str = NULL;
      int error = ((OutputObject *)(self))->error;
  
!     if (!PyArg_ParseTuple(args, "es#", p_enc, &str, &len))
  	return NULL;
  
      Py_BEGIN_ALLOW_THREADS
--- 74,80 ----
      char *str = NULL;
      int error = ((OutputObject *)(self))->error;
  
!     if (!PyArg_ParseTuple(args, "es#", ENC_OPT, &str, &len))
  	return NULL;
  
      Py_BEGIN_ALLOW_THREADS
***************
*** 108,114 ****
  	char *str = NULL;
  	PyInt len;
  
! 	if (!PyArg_Parse(line, "es#", p_enc, &str, &len)) {
  	    PyErr_SetString(PyExc_TypeError, _("writelines() requires list of strings"));
  	    Py_DECREF(list);
  	    return NULL;
--- 114,120 ----
  	char *str = NULL;
  	PyInt len;
  
! 	if (!PyArg_Parse(line, "es#", ENC_OPT, &str, &len)) {
  	    PyErr_SetString(PyExc_TypeError, _("writelines() requires list of strings"));
  	    Py_DECREF(list);
  	    return NULL;
*** ../vim-7.3.231/src/version.c	2011-06-26 03:16:58.000000000 +0200
--- src/version.c	2011-06-26 04:00:40.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     232,
  /**/

-- 
If evolution theories are correct, humans will soon grow a third
hand for operating the mouse.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
